### PR TITLE
Add header

### PR DESCRIPTION
**Pull request by carlganz**
Let users add fields to the jwt header

### CODE REVIEW

**Comment by jeroen** [comment]
Thanks! Does jose allow custom fields in the header? Can you point me to the spec or examples that this is allowed?

File: R/jwt.R [hunk 1]
```diff
@@ -34,18 +35,25 @@
...
     secret <- charToRaw(secret)
   if(!is.raw(secret))
     stop("Secret must be a string or raw vector")
   if(inherits(secret, "rsa") || inherits(secret, "dsa") || inherits(secret, "ecdsa"))
     stop("Secret must be raw bytes, not a: ", class(secret)[-1])
-  header <- to_json(list(
-    typ = "JWT",
-    alg = paste0("HS", size)
-  ))
+  header <- if (is.null(header)) {
```
Comment (jeroen) [inline R/jwt.R]: I think if/else is redundant? If `header == NULL` then `c(list(...), header)` below will naturally reduce to just `list(...)` ?

Comment (carlganz) [reply R/jwt.R]: You are right

**Comment by jeroen** [comment]
Ok thanks. You mean the `kid` parameter, correct? It seems that [kid is a standardized jwt parameter](https://tools.ietf.org/html/rfc7515#section-9.1.2). I'm still not sure if the jwt spec allows for including arbitrary parameters or only the ones [listed here](https://tools.ietf.org/html/rfc7515#section-9.1.2)? Are there also examples that include custom application parameters?

Thanks for your patience :-)

**Comment by carlganz** [comment]
Yes kid parameter is what I need. Looks like you are right that arbitrary data is not allowed in header. 

How should header parameters be enforced?

**Comment by jeroen** [comment]
I think it's fine for now. I might tune it a bit later.

**Comment by jeroen** [comment]
I have changed the syntax a bit. You now need to pas the parameters directly instead of via a `header` argument:

```r
jwt <- jwt_encode_hmac(jwt_claim(test = "test"), secret = key, test = "test")
```

**Comment by jeroen** [comment]
Actually never mind let's keep it at your implementation :)

**Comment by jeroen** [comment]
This is on CRAN now.

**Pull request merged**